### DISCUSSION TITLE
It is not possible to move the cursor outside the active buffer using ANSI commands

### DISCUSSION THREAD

**Discussion by DevTester425** [sourcelair/xterm.js#5042] [Q&A] [answered]
Please describe the solution how to move the cursor to position **1;1** if **term.buffer.active.baseY** > **0**.
If this cannot be done by **ANSI commands**, describe any method.

## Details
- Browser and browser version: all 
- OS version: all
- xterm.js version: 5.1.0

### Steps to reproduce
run code
```
<!doctype html>
  <html>
    <head>
      <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/xterm@5.1.0/css/xterm.min.css" />
      <script src="https://cdn.jsdelivr.net/npm/xterm@5.1.0/lib/xterm.min.js"></script>
    </head>
    <body>
      <div id="terminal"></div>
      <script>
        (async () => {
            var term = new Terminal({
                rows: 10
            });
            term.open(document.getElementById('terminal'));
            await write(term, '\r\n1\r\n2\r\n3\r\n4\r\n5\r\n6\r\n7\r\n8\r\n9\r\n10\r\n11\r\n12\r\n13\r\n14\r\n15');
            term.scrollToTop();
            await write(term, '\u001b[1;1H');
 
            async function write(term, data) {
                return new Promise((resolve) => {
                    term.write(data, resolve)
                })
            }
        })()
      </script>
    </body>
  </html>
```
Result
![зображення](https://github.com/xtermjs/xterm.js/assets/88549983/f6e3104d-c95a-4a32-b381-49d5c2245b73)

**Comment by Tyriar** [comment, marked as answer]
This is by design, once there's content to scroll you cannot change it outside of clearing it entirely.

**Reply by DevTester425** [reply]
Thanks for the answer.
This behavior is different from the behavior of terminals.
Is there no way to achieve this?

**Reply by jerch** [reply]
> This behavior is different from the behavior of terminals.

Nope, thats exactly as all xterm-like terminal emulators behave.

> Is there no way to achieve this?

Not an easy one. The hard one would be to fork the repo and expand the cursor logic to the full scrollbuffer yourself. We wont do that, as its not how it is intended to work in terminals.

**Reply by DevTester425** [reply]
Thanks for the answer.
How about the approach of resizing the terminal every time the content doesn't fit in the active buffer?
See the code below.
```
<!doctype html>
  <html>
    <head>
      <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/xterm@5.1.0/css/xterm.min.css" />
      <script src="https://cdn.jsdelivr.net/npm/xterm@5.1.0/lib/xterm.min.js"></script>
    </head>
    <body>
      <div id="terminal"></div>
      <script>
        (async () => {
            var term = new Terminal({
              rows: 10
            });
            term.open(document.getElementById('terminal'));
            
            for (let i = 0; i < 100; i++) {
              await write(term, '\r\n' + (i + 1));
            }
            await write(term, '\u001b[1;1H');

            async function write(term, data) {
                return new Promise((resolve) => {
                  term.write(data, resolve);
                  if (term.buffer.active.baseY > 0) {
                    term.resize(term.cols, term.rows + term.buffer.active.baseY);
                  }
                })
            }
            })()
      </script>
    </body>
  </html>
```

**Reply by Tyriar** [reply]
You could do that, but you're going to run into performance problems eventually if it gets big enough

**Reply by DevTester425** [reply]
Thanks for the answer.
